### DISCUSSION TITLE
CPU consumption 100% on Raspberry PI 4

### DISCUSSION THREAD

**Discussion by demetera** [coder/code-server#5929] [Q&A]
Dear community,

installed code-server from tar.gz package - [code-server-4.9.1-linux-arm64.tar.gz](https://github.com/coder/code-server/releases/download/v4.9.1/code-server-4.9.1-linux-arm64.tar.gz)

Unpacked to /opt/code-server

Permanent CPU consumption 100%. No any active extensions installed. See the screenshot:

![image](https://user-images.githubusercontent.com/3475944/209709682-1f3ed348-e84b-4a0f-a351-d35030090ecc.png)

Any suggestions, how to tackle?

Thank you in advance

**Comment by demetera** [comment]
Update: I've installed via npm (as recommended by the team) and everything working fine at the moment. No CPU activity in idle mode at all. But I'm curious to understand, why it was happening with tar.gz installation

**Comment by lg-nineoneninethreenine** [comment]
IMO, that is the first time I see that! I have a Pi 4 too, I will check and report my findings if I can. Maintainers are off for Christmas, so you could expect less than usual level of help here.

**Reply by demetera** [reply]
Thanks for your reply! Just an observation. That's why I didn't post it as an issue. I have Umbrel on the same machine, but I've disabled all containers and services. Need to test on clean image as well

**Reply by lg-nineoneninethreenine** [reply]
I wonder if the tarball is doing too many processes. Do you have any extensions that work round the clock? Perhaps that could explain the issue.

**Reply by lg-nineoneninethreenine** [reply]
Hmm, no extensions... That is the weirdest thing I have seen. I will be out for vacation, but will be checking in often. Can you try getting a bit more data on the Pi? I think the tarball is doing too much work hence the CPU is being loaded.

**Reply by demetera** [reply]
Yes, that was very weird situation. Today I've reinstalled Raspberry PI and CPU isn't loaded at all anymore. Only a little, when I open URL in browser. But I will monitor further. Thanks for the help!

**Comment by demetera** [comment]
The problem was caused  by another operating system installed (Umbrel OS - based on Raspberry PI OS). Even I've stopped all docker containers and additional services (as I could), built-in nodejs or npm or whatever,  unexpected CPU overload when running code-server tarball doesn't take a place anymore
